### ISSUE TITLE
`prefer_final_locals` false positive in for-loop destructured records

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#4539]
See: https://github.com/dart-lang/sdk/issues/52820

**TL;DR:** the lint should only fire if all declared variables can be changed to final

Repro:

```dart
f() {
  for (var (a, b) in [(1, 2)]) {
    ++a;
  }
}
```

(Should not flag `b`.)
